java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -5257,4 +5265,54 @@ public void blmove_returns_success() {
         assertEquals(testResponse, response);
         assertEquals(value, payload);
     }
+
+    @SneakyThrows
+    @Test
+    public void bitfieldReadOnly_returns_success() {
+        // setup
+        String key = "testKey";
+        Long[] result = new Long[] {7L};
+        BitFieldGet subcommand = new BitFieldGet(new SignedEncoding(4), new Offset(2));
+        String[] args = ArrayUtils.addFirst(subcommand.toArgs(), key);
+        CompletableFuture<Long[]> testResponse = new CompletableFuture<>();
+        testResponse.complete(result);
+
+        // match on protobuf request
+        when(commandManager.<Long[]>submitNewCommand(eq(BitFieldReadOnly), eq(args), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<Long[]> response =
+                service.bitfieldReadOnly(key, new BitFieldReadOnlySubCommands[] {subcommand});
+        Long[] payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(result, payload);
+    }
+
+    @SneakyThrows
+    @Test
+    public void bitfield_returns_success() {
+        // setup
+        String key = "testKey";
+        Long[] result = new Long[] {7L};
+        BitFieldSet subcommand = new BitFieldSet(new SignedEncoding(4), new Offset(2), 3);
```

Comment:
we should test more options, to make sure that we're creating the correct string[] array to be sent to Redis in the proper order.  Ideally, we should test each constant in a unit test (and make sure OVERFLOW, GET, SET, INCRBY are correctly included in the `submitNewCommand` call).